### ISSUE TITLE
Validators for Bytes fields

### ISSUE THREAD

**Issue by joewreschnig** [ent/ent#1714]
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

We would like to configure validators for `Bytes` fields similar to string fields. Introducing a new method,
```
// Validate adds a validator for this field. Operation fails if the validation fails.
func (b *bytesBuilder) Validate(fn func([]byte) error) *bytesBuilder {
	b.desc.Validators = append(b.desc.Validators, fn)
	return b
}
```
would be sufficient for us. (It even seems to codegen something reasonable already when used, but I didn't run any tests to see if it's correct.)

We would also be able to use similar `MinLen`, `MaxLen`, and `NotEmpty` default validators. For our use case it would also be best if `bytesBuilder#MaxLen` also configured a validator, but since this would break compatibility I understand if that can't be done. In that case I would suggest only providing `Validate` as it would be confusing if `MaxLen` affected the descriptor and not the validator, but `MinLen` the validator and not the descriptor.

## Motivation 🔦

We have some binary blobs which have some length requirements (e.g. not empty and < 20kb) we would like to enforce via ent's mutators and not ad hoc validation.

**Comment by a8m** [comment]
Hey @joewreschnig and thanks for proposing this feature. I added support for the `Validator` API (see https://github.com/ent/ent/pull/1715) - will merge it soon.

> We would also be able to use similar MinLen, MaxLen, and NotEmpty default validators.

I think those helpers can be useful. Let's leave this issue open and get feedback from additional users 😃 

You're welcome to join our [Slack channel](https://entgo.io/docs/slack/) btw 😎

**Comment by a8m** [comment]
Closed via #1863, #1867 and #1869

Thanks @hedwigz for your contribution 🙏
